editor/src/components/canvas/canvas-strategies/strategies/flex-resize-strategy.tsx

```diff
@@ -0,0 +1,426 @@
+                ),
+              )
+            }
+          }
+          if (snapToParentEdge != null && !snapToParentEdge.snap) {
+            resizeCommands.push(
+              deleteProperties('always', selectedElement, [
+                stylePropPathMappingFn('flexGrow', styleStringInArray),
+              ]),
+            )
+          }
+
+          return strategyApplicationResult([
+            ...resizeCommands,
+            updateHighlightedViews('mid-interaction', []),
+            setCursorCommand(pickCursorFromEdgePosition(edgePosition)),
+            setElementsToRerenderCommand(selectedElements),
+          ])
+        } else {
+          return strategyApplicationResult([
+            updateHighlightedViews('mid-interaction', []),
+            setCursorCommand(pickCursorFromEdgePosition(edgePosition)),
+          ])
+        }
+      }
+      // Fallback for when the checks above are not satisfied.
+      return emptyStrategyApplicationResult
+    },
+  }
+}
+
+function shouldSnapToParentEdge(
+  edgePosition: EdgePosition,
+  resizedBounds: CanvasRectangle,
+  parentBounds: CanvasRectangle | null,
+  parentFlexDirection: FlexDirection | null,
+  element: ElementInstanceMetadata,
+  startingMetadata: ElementInstanceMetadataMap,
+): {
+  snapDirection: 'horizontal' | 'vertical'
+  snap: boolean
+} | null {
+  const parentPadding = element.specialSizeMeasurements.parentPadding
+  const parentJustifyContent = element.specialSizeMeasurements.parentJustifyContent
+  const parentGap = element.specialSizeMeasurements.parentFlexGap
+
+  const flexSiblingsWithoutSelected = MetadataUtils.getSiblingsUnordered(
+    startingMetadata,
+    element.elementPath,
+  )
+    .filter((sibling) => !EP.pathsEqual(sibling.elementPath, element.elementPath))
+    .filter(MetadataUtils.elementParticipatesInAutoLayout)
+
+  const anySiblingFillSized = flexSiblingsWithoutSelected.some((sibling) => {
+    const fillHugFixedState = detectFillHugFixedState(
+      parentFlexDirection === 'row' ? 'horizontal' : 'vertical',
+      startingMetadata,
+      sibling.elementPath,
+    )
+    return fillHugFixedState?.type === 'fill'
+  })
+
+  if (anySiblingFillSized) {
+    return null
+  }
+
+  if (parentFlexDirection === 'row-reverse' || parentFlexDirection === 'column-reverse') {
+    return null
+  }
+
+  if (parentBounds == null) {
+    return null
+  }
+
+  const siblingFrames = flexSiblingsWithoutSelected.map((sibling) =>
+    MetadataUtils.getFrameInCanvasCoords(sibling.elementPath, startingMetadata),
+  )
+
+  const parentInnerBounds = canvasRectangle({
+    x: parentBounds.x + (parentPadding.left ?? 0),
+    y: parentBounds.y + (parentPadding.top ?? 0),
+    width: parentBounds.width - ((parentPadding.left ?? 0) + (parentPadding.right ?? 0)),
+    height: parentBounds.height - ((parentPadding.top ?? 0) + (parentPadding.bottom ?? 0)),
+  })
+
+  if (parentFlexDirection === 'row' && edgePosition.x !== 0.5) {
```

Comment:
I think this portion of code could do with some comments to outline what the actual snapping rules are here

Reply:
updated with comments